tests/integration/targets/ec2_eip/tasks/main.yml

```diff
@@ -520,11 +548,37 @@
     ec2_eni:
       state: absent
       eni_id: '{{ eni_create_a.interface.id }}'
+  - name: Make sure the instance is ready
+    ec2_instance_info:
+      filters:
+        "tag:Name": '{{ resource_prefix }}-instance'
+    register: instance_info
+    until: instance_info.instances[0].state.name == 'running'
+  - name: Attach eip to an EC2 instance
+    ec2_eip:
+      device_id: '{{ instance_info.instances[0].instance_id }}'
+      state: present
+      release_on_disassociation: yes
+    register: instance_eip
+  - assert:
+      that:
+        - instance_eip is success
+        - instance_eip.allocation_id is defined
+        - instance_eip.release_on_disassociation == True
```

Comment:
Looks like release_on_disassociation isn't available:

```
05:03 changed: [testhost] => {"allocation_id": "eipalloc-0ad299071a4e465d1", "changed": true, "public_ip": "52.44.77.215", "resource_actions": ["ec2:DescribeAddresses", "ec2:DescribeInstances", "ec2:AssociateAddress", "ec2:AllocateAddress"]}
```

Might be worth throwing an ec2_eip_info call in there.